tests/Api/Parser/JsonParserTest.php

```diff
@@ -15,116 +15,96 @@
      */
-    public function testHandlesTimeStampsWhenUnknownType($timestamp, $expectedValue)
+    public function testHandlesTimeStampsWhenIso8601Expected($timestamp, $commandName, $expectedValue)
     {
         $service = $this->generateTestService();
-        $client = $this->generateTestClient($service, ['UnknownTimestamp' => $timestamp]);
-        $command = $client->getCommand('ParseJson');
+        $client = $this->generateTestClient($service, ['Timestamp' => $timestamp]);
+        $command = $client->getCommand($commandName);
         $list = $client->getHandlerList();
         $handler = $list->resolve();
-        $result = $handler($command)->wait()['UnknownTimestamp']->__toString();
+        $result = $handler($command)->wait()['Timestamp']->__toString();
         self::assertEquals($expectedValue, $result);
     }
 
     /**
      * @dataProvider timeStampExceptionModelProvider
      */
-    public function testHandlesTimeStampWhenIso8601ExpectedThrowsException($timestamp, $expectedException, $expectedMessage)
+    public function testTimeStampExceptions($timestamp, $commandName, $expectedException, $expectedMessage)
     {
         $service = $this->generateTestService();
-        $client = $this->generateTestClient($service, ['Iso8601Timestamp' => $timestamp]);
-        $command = $client->getCommand('ParseJson');
+        $client = $this->generateTestClient($service, ['Timestamp' => $timestamp]);
+        $command = $client->getCommand($commandName);
         $list = $client->getHandlerList();
         $handler = $list->resolve();
         $this->setExpectedException($expectedException, $expectedMessage);
         $handler($command)->wait();
     }
 
-    /**
-     * @dataProvider timeStampExceptionModelProvider
-     */
-    public function testHandlesTimeStampWhenUnixExpectedThrowsException($timestamp, $expectedException, $expectedMessage)
-    {
-        $service = $this->generateTestService();
-        $client = $this->generateTestClient($service, ['UnixTimestamp' => $timestamp]);
-        $command = $client->getCommand('ParseJson');
-        $list = $client->getHandlerList();
-        $handler = $list->resolve();
-        $this->setExpectedException($expectedException, $expectedMessage);
-        $handler($command)->wait();
-    }
-
-    /**
-     * @dataProvider timeStampExceptionModelProvider
-     */
-    public function testHandlesTimeStampWhenUnknownTypeThrowsException($timestamp, $expectedException, $expectedMessage)
-    {
-        $service = $this->generateTestService();
-        $client = $this->generateTestClient($service, ['UnknownTimestamp' => $timestamp]);
-        $command = $client->getCommand('ParseJson');
-        $list = $client->getHandlerList();
-        $handler = $list->resolve();
-        $this->setExpectedException($expectedException, $expectedMessage);
-        $handler($command)->wait();
-    }
 
     private function generateTestClient(Service $service, $args = [])
```

Comment:
To avoid repetition, could the test client and test service code be shared as a trait across the `JsonParserTest` and `XmlParserTest`?

If the only difference is the metadata `protocol` value, could that be set from a method parameter if it's shared across the two test classes?

Reply:
It's not the only difference.  They have some different input tests because of how the two protocols store values differently.  We could potentially combine them and have some tests that are shared between the two and some that are protocol specific using three data providers.  Would you rather that?

Reply:
Yes; what would be a good name for the trait class?  ParserTestTrait?

Reply:
`ParserTestServiceTrait` perhaps?

Reply:
Great, I just renamed it and pushed it.  Thank you for the suggestion